02_cleaning.Rmd

```diff
@@ -1035,3 +1035,212 @@ print(objSDC, "risk")
+  #' @param language_codes Character vector of two-letter language codes. The input vector will be translated from both of these languages.
+  #' @param target_lang Input vector will be translated into this language.
+  #' @param threshold Input threshold to interrupt the user if the number of characters is exceeding 200,000 by default. 
+  #' @returns The same dataframe as `responses`, but with a new column, containing the translation.
+  #' The column will be named according to the given source and target languages. By default, the output will be stored in column named 'response.en.from.uk'
+  info_df <- data.frame()
+  responses_batch <- data.frame()
+  temp_resp_whole <- data.frame()
+  start_time <- Sys.time()
+
+  # extract unique responses from the source dataframe
+  responses <- responses %>% mutate(resp_lower = str_to_lower(!!sym(values_from)))
+   
+  input_vec <- responses %>% distinct(resp_lower) %>% pull(resp_lower)
+  # cleaning up html leftovers:
+  input_vec <- gsub("&#39;", "'", input_vec)
+  # counts characters which will be translated
+  char_counter <- sum(str_length(input_vec))
+  # TODO: pause here, print the char_counter, and ask the user if the translation should go ahead
+  if (char_counter > threshold){
+    yes_no <- svDialogs::dlgInput(paste0("The number of characters exceeds ", threshold, ". Please enter [YES] if you would like to proceed or [NO] to kill:"), "YES or NO")$res
+  } else{
+    yes_no <- "YES"
+  }
+  batching <- svDialogs::dlgInput(paste0("How many batches would you like to split your translation (",char_counter," characters)? (please only integer)"), 0)$res
+  batching <- as.numeric(batching)
+  if(yes_no == "YES"){
+    if(length(input_vec) > 0){
+      for (code in language_codes) {
+        col_name <- paste0("response.",target_lang, ".from.",code)
+        temp_resp <- tibble(input_vec)
+        temp_resp[[col_name]] <- NA
+        temp_resp <-  temp_resp[sample(1:nrow(temp_resp)),]
+        ## create batches
+        temp_resp_batches <- split(temp_resp, factor(sort(rank(row.names(temp_resp))%%batching)))
+        progress.bar.title <- as.character(Sys.time())
+        pb <- tcltk::tkProgressBar(progress.bar.title, "Number of batches executed", 0, batching, 0, width = 600)
+        prog <- 1
+        for (temp_resp_batch in temp_resp_batches){
+          tcltk::setTkProgressBar(pb, prog, progress.bar.title, paste0("Number of batches executed: ", prog, " of ", batching,"\n",length(temp_resp_batch$input_vec)," responses will be translated from ",code," to ",target_lang, "\nThis means ",sum(str_length(temp_resp_batch$input_vec))," utf-8 characters."))
+          prog <- prog + 1
+          # actual translation:
+          result_vec <- NULL
+          result_vec <- try(translateR::translate(content.vec = temp_resp_batch$input_vec,
+                              microsoft.api.key = source("resources/microsoft.api.key_ukraine.R")$value,
```

Comment:
Can we make it more general to enter the Microsoft API key for any mission, rather than being limited to the Ukraine API key